# Skyline Enrollments: Confidence Intervals

Lesson 1.5 Practice. Computes confidence intervals for means and
proportions on the Skyline Enrollments synthetic dataset, plus a simulation
demonstrating the 95% interpretation.

Author: [A. Chris Yi]
Date: [2026-09-23]

In [6]:
import numpy as np
import pandas as pd
from scipy import stats
import plotly.express as px

In [19]:
enrollments = pd.read_csv("../lesson-1-2-types-of-data/skyline_enrollments.csv")

n = len(enrollments)
sample_mean = enrollments["hours_studied"].mean()
sample_std = enrollments["hours_studied"].std()

print(f"Sample size: n = {n}")
print(f"Sample mean: {sample_mean}")
print(f"Sample standard deviation: {sample_std}")



Sample size: n = 100
Sample mean: 30.63165105664007
Sample standard deviation: 15.99445895284144


In [20]:
standard_error = sample_std / np.sqrt(n)
z = 1.96  # for a 95% confidence interval
margin_of_error = z * standard_error
confidence_interval = (sample_mean - margin_of_error, sample_mean + margin_of_error)
print(f"95% Confidence Interval: {confidence_interval}")
print(f"Margin of Error: {margin_of_error}")
print(f"Standard Error: {standard_error}")


95% Confidence Interval: (np.float64(27.49673710188315), np.float64(33.76656501139699))
Margin of Error: 3.134913954756922
Standard Error: 1.599445895284144


The average hours studied in the sample is 30.6 hours. Based on the sample size of 100 students and a sample standard deviation of 15.99, a 95% confidence interval for the true mean is 27.50 to 33.77 hours. With 95% confidence, the true average falls somwhere in that range. 

In [25]:
n = len(enrollments)
n_completion = len(enrollments[enrollments["completion_status"] == "Completed"])
se_prop = np.sqrt(p_hat * (1 - p_hat) / n)
z = 1.96  # for a 95% confidence interval
moe_prop = z * se_prop
ci_prop_lower = p_hat - moe_prop
ci_prop_upper = p_hat + moe_prop
ci_prop = (ci_prop_lower, ci_prop_upper)

print(f"Sample Size: {n}")
print(f"Number of Completions: {n_completion}")
p_hat = n_completion / n
print(f"Proportion of Completions: {p_hat}")

print(f"Standard Error: {se_prop}")
print(f"Margin of Error: {moe_prop}")
print(f"95% Confidence Interval: {ci_prop}")


Sample Size: 100
Number of Completions: 72
Proportion of Completions: 0.72
Standard Error: 0.0448998886412873
Margin of Error: 0.08800378173692311
95% Confidence Interval: (np.float64(0.6319962182630768), np.float64(0.8080037817369231))


The proportion of completions in this sample is 72% with a margin of error of 8.8%. Future samples should show completions between 63% and 81% at the current sample size. 

In [ ]:
The confidence interval in this sample is rather wide at 8.8%.In order to halve the margin of error, you would need to increase the sample size by a factor of eight, since the margin of error is inversely proportional to the square root of the sample size. I would suggest a sample of 800 participants (approximately 8x the current sample size) to get the margin of error to +- 2%.